code/ship/shiphit.cpp

```diff
@@ -2448,14 +2448,33 @@ static void ship_do_damage(object *ship_objp, object *other_obj, vec3d *hitpos,
 				apply_diff_scale = false;
 			}
 		}
+
+		// if weapon is vampiric, slap healing on shooter instead of target
+		weapon_info_index = shiphit_get_damage_weapon(other_obj);
+		if (weapon_info_index >= 0) {
```

Comment:
So, one style thing.  You'll want to put a new line before an if statement to make it distinct from surrounding code.  Also after if there's code after it, too, but this one is just curly braces after